spec/models/package_manager/maven_spec.rb

```diff
@@ -206,6 +221,13 @@
       it "to find repository url" do
         expect(parsed[:repository_url]).to eq("https://github.com/googleapis/googleapis-dummy")
       end
+
+      it "to map properties" do
+        expect(parsed[:properties]).to include(
+          "api.version" => "1.0.0-rc1",
+          "scm.url" => "https://github.com/googleapis/googleapis-dummy"
```

Comment:
This spec fails because this value is not resolved because our code seems to prefer selecting the child POM values. I can't tell if our code is wrong or the spec setup is wrong in here. The parent/child property resolution seems to work correctly when I run the code with live data from Maven Central and the setup in the spec looks like it is what I would expect so I'm a little stumped on what is going on.

I'm wondering if this line is backwards logic wise and it should actually prefer the parent property?
https://github.com/librariesio/libraries.io/blob/87a040a55ad04ff379c056103accf3bf7a8523cd/app/models/package_manager/maven.rb#L149

Reply:
I'm not sure I understand the issue with this spec here either. This spec aside I am more than happy to approve though. If you do go through with your suggestion and swap the logic, do the specs pass appropriately?

Reply:
I'm pretty sure the child is preferred in Maven, but I can test that theory right now.

Reply:
I would always expect the child to reference the parent via `${propertyName}` and the spec seems to always return the child's unresolved property name when referenced. There may be some code being skipped by the mocked returns?

Reply:
confirmed that with this child pom's `scm.url`: https://github.com/tiegz/grandiorite/blob/main/java/maven/pom.xml#L18-L22
and this parent pom's `scm.url`: https://github.com/tiegz/grandiorite/blob/main/java/maven/grandiorite-base/pom.xml#L23-L27. When I run `mvn help:evaluate`, then `${project.scm.url}` returns the child's url.